threatingestor/sources/github.py

```diff
@@ -8,29 +8,56 @@
 SEARCH_URL = "https://api.github.com/search/repositories"
 
 class Plugin(Source):
-
-    def __init__(self, name, search):
+    """Github Source Plugin"""
+    def __init__(self, name, search, username="", token=""):
         self.name = name
         self.search = search
 
+        if username and token:
+            self.auth = (username, token)
+        else:
+            self.auth = None
+
+    def make_requests(self, url, params, auth):
+        """Returns a list of artifacts"""
+        # Iterates through pages of results from query
+        artifact_list = []
+
+        response = requests.get(url, params=params, auth=self.auth)
+        while True:
+            for repo in response.json()['items']:
```

Comment:
This will raise an exception if the GitHub API response is something we don't expect. We could try/except, but I think a cleaner way is just to fail gracefully and loop over an empty list.

```suggestion
            for repo in response.json().get('items', []):
```

Though that doesn't handle the case where `response.json()` fails and throws a `simplejson.errors.JSONDecodeError`, if that response isn't able to be parsed as JSON. I can't find anything in the GitHub API docs that says how it handles 500 errors - we might get a standard HTML response in those cases.